test/unit/wallet.js

```diff
@@ -203,6 +203,37 @@ describe('Wallet Prototype Methods', function() {
       bgUrl = common.Environments[bitgo.getEnv()].uri;
     }));
 
+    it('extra unspent fetch params', co(function *() {
+      const billingAddress = '2MswQjkvN6oWYdE7L2brJ5cAAMjPmG59oco';
+      const customUnspentsFetchParams = { test: 123 };
+      const sendAmount = 1e5;
+
+      nock(bgUrl)
+      .post('/api/v1/billing/address')
+      .reply(200, { address: billingAddress });
+
+      const scope = nock(bgUrl)
+      .get(`/api/v1/wallet/${fakeWallet.id()}/unspents`)
+      .query(_.merge(customUnspentsFetchParams, {
+        segwit: true,
+        target: sendAmount,
+        minSize: 0
+      }))
+      .reply(200, { unspents: [] });
+
+      yield fakeWallet.createTransaction({
+        unspentsFetchParams: customUnspentsFetchParams,
+        recipients: { [billingAddress]: sendAmount },
+        feeRate: 10000,
+        bitgoFee: {
+          amount: 0,
+          address: ''
+        }
+      }).should.be.rejectedWith('0 unspents available for transaction creation');
+
+      scope.isDone().should.be.true();
+    }));
+
```

Comment:
Why isn't `allowLedgerSegwit` exercised in this test?

Reply:
Because it's testing whether or not `createTransaction()` correctly passes custom unspent parameters when it makes a call to get unspents. What the exact set of custom unspent parameters is not relevant (I chose `{ test: 123 }` in this case). We are just trying to make sure that param is given to the server.